File: platforms/macos/PwdlockMac/Sources/PwdlockCore/Vault/VaultBootstrap.swift
import Foundation

public struct CreatedVault: Sendable {
    public let metadata: VaultMetadata
    public let vaultKey: Data
}

public enum VaultBootstrap {
    public static func create(masterPassword: String) throws -> CreatedVault {
        let vaultKey = try SecureRandom.bytes(count: 32)
        let metadata = try VaultKeyEnvelope.wrap(
            vaultKey: vaultKey,
            masterPassword: masterPassword,
            vaultID: SecureRandom.bytes(count: 16),
            passwordSalt: SecureRandom.bytes(count: 16),
            wrapNonce: SecureRandom.bytes(count: 12),
            parameters: .initial
        )
        return CreatedVault(metadata: metadata, vaultKey: vaultKey)
    }
}

File: platforms/android/app/src/main/java/com/pwdlock/android/crypto/VaultBootstrap.kt
package com.pwdlock.android.crypto

/**
 * 复刻 macOS `VaultBootstrap.create`：创建新保险库。
 * 生成随机 32 字节 vaultKey 与随机 vaultID / passwordSalt / wrapNonce，
 * 用主密码包装成 [VaultMetadata]。
 */
object VaultBootstrap {
    data class CreatedVault(val metadata: VaultMetadata, val vaultKey: ByteArray)

    fun create(masterPassword: String): CreatedVault {
        val vaultKey = CryptoIO.randomBytes(32)
        val metadata = VaultKeyEnvelope.wrap(
            vaultKey = vaultKey,
            masterPassword = masterPassword,
            vaultID = CryptoIO.randomBytes(16),
            passwordSalt = CryptoIO.randomBytes(16),
            wrapNonce = CryptoIO.randomBytes(12),
        )
        return CreatedVault(metadata, vaultKey)
    }
}
